### PR TITLE
Support authentication backends that don't have usernames

### LINKED ISSUE #31: Does not work with OAuth 2 bearer tokens
When used with OAuth 2 bearer tokens as provided by [django-oauth-toolkit](https://github.com/evonove/django-oauth-toolkit), a bad or missing token causes a 500 server error instead of a 403 unauthorized as expected:

```
Internal Server Error: /v2/folders/all
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py", line 108, in get_response
    response = middleware_method(request)
  File "/usr/local/lib/python2.7/dist-packages/oauth2_provider/middleware.py", line 38, in process_request
    user = authenticate(request=request)
  File "/usr/local/lib/python2.7/dist-packages/django/contrib/auth/__init__.py", line 74, in authenticate
    user = backend.authenticate(**credentials)
  File "/usr/local/lib/python2.7/dist-packages/ratelimitbackend/backends.py", line 25, in authenticate
    username = kwargs[self.username_key]
KeyError: 'username'
```

I already have a fix for this so I'll open a PR.

### PR DESCRIPTION
**Pull request by scjody**
Fixes #31, includes tests.

### CODE REVIEW

**Comment by scjody** [comment]
@brutasse This is ready to go. Please let me know what you think!

File: ratelimitbackend/backends.py [hunk 1]
```diff
@@ -19,9 +19,13 @@ class RateLimitMixin(object):
     minutes = 5
     requests = 30
     username_key = 'username'
+    no_username = False
 
     def authenticate(self, request=None, **kwargs):
-        username = kwargs[self.username_key]
+        username = kwargs.get(self.username_key)
+        if username is None and not self.no_username:
+            raise KeyError
```
Comment (brutasse) [inline ratelimitbackend/backends.py]: Would this achieve the same thing?

```python
username = None
try:
    username = kwargs[self.username_key]
except KeyError:
    if not self.no_username:
        raise
```

with the benefit of raising KeyError with the relevant key.

Comment (scjody) [inline ratelimitbackend/backends.py]: Yes, that's reasonable, and I've made the change.

**Comment by scjody** [comment]
@brutasse Thanks for the review! Your change sounds good so I made it.

**Comment by scjody** [comment]
@brutasse Do you have any idea why the build is failing? The error is:
```
ERROR: InterpreterNotFound: python3.3
```
I haven't made any changes that would impact this, and my initial PR about a week ago passed CI.

**Comment by scjody** [comment]
@brutasse I rebased on your master and CI passes now. I think this is ready to merge but please let me know if you would like any more changes.

**Comment by brutasse** [comment]
Great, thanks @scjody! Sorry for the silence on the build issue… I didn't know 3.3 now has to be pulled manually.

**Comment by brutasse** [comment]
@scjody version 1.2 is up on PyPI with these changes.

**Pull request merged**